src/components/ArticleDetails.jsx

```diff
@@ -1,22 +1,57 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { getArticleById } from "../api";
+import { getArticleById, updateVoteByArticleId } from "../api";
 import { formatAPIDate } from "./Utils";
 import CommentsList from "./CommentsList";
+import Loading from "./Loading";
+import { FaRegThumbsUp, FaRegThumbsDown } from "react-icons/fa";
 
 function ArticleDetails() {
-  const [article, setArticle] = useState({});
+  const [article, setArticle] = useState(null);
+  const [hasVoted, setHasVoted] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
   const { article_id } = useParams();
 
   useEffect(() => {
+    setIsLoading(true);
     getArticleById(article_id)
       .then((response) => {
         setArticle(response.data.article);
+        setIsLoading(false);
       })
       .catch((error) => {
         console.log(error);
+        setIsLoading(false);
       });
-  }, []);
+  }, [article_id]);
+
+  const handleVote = (inc_votes) => {
+    if (!hasVoted) {
+      setArticle((prevArticle) => ({
+        ...prevArticle,
+        votes: prevArticle.votes + inc_votes,
+      }));
+      setHasVoted(true);
+
+      updateVoteByArticleId(article_id, inc_votes)
+        .then((response) => {})
+        .catch((error) => {
+          setArticle((prevArticle) => ({
+            ...prevArticle,
+            votes: prevArticle.votes - inc_votes,
+          }));
+          setHasVoted(false);
+          console.log(error);
+          alert("Failed to vote. Please try again.");
```

Comment:
This is doing a great job in terms of making it clear something has gone wrong and resetting the votes - but alerts are quite notorious for being annoying and bad user experience - could you think of an alternate way to show this error message?